### PR TITLE
Trim whitespace when setting display names

### PR DESCRIPTION
**Pull request by mohit-rathee**
### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [x] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). 
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters)
### Updated the code and written a test too. 
This code is not exploitable by the exploit i said earlier when user send only spaces in thier displayname.

### CODE REVIEW

**Comment by erikjohnston** [comment]
Thanks @mohit-rathee! Can you please add a newsfile and sign off this PR as per the [contributing guide](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#9-submit-your-patch) please? The change log can be as simple as a "Trim any white space when setting display names."

**Comment by mohit-rathee** [comment]
I believe that i have made the changes:
1. Adding changelog.
2. Adding sign-off

And, thanks for your support and guidance during my contributions to the repository.

And i should mention that I look forward to continue my contribution for this open-source community in the future.

File: changelog.d/16031.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Trim any white space when setting display names.
```
Comment (clokep) [inline changelog.d/16031.bugfix:1]:
```suggestion
Remove leading and trailing spaces when setting a display name.
```

**Review by clokep — APPROVED** [review]
LGTM if tests pass!

**Pull request merged**